scripts/prepare-machine.ps1

```diff
@@ -64,6 +64,14 @@ $ClogDownloadUrl = "https://github.com/microsoft/CLOG/releases/download/v$ClogVe
 
 $MessagesAtEnd = New-Object Collections.Generic.List[string]
 
+# Detect PS by feature
+if (!(Test-Path Variable:\IsWindows)) {
```

Comment:
I think I would prefer an explicit version check rather then just checking for that variable. Otherwise pwsh 6 would work, which we don't do any testing on.

Just checking that $PSVersionTable.PSVersion.Major is >= 7 would probably be the best.

Reply:
> `Otherwise pwsh 6 would work, which we don't do any testing on.` 

Makes sense if it's not tested there. Updated.